src/devices/Ht16k33/SevenSegment.cs

```diff
@@ -0,0 +1,235 @@
+        public void WriteHex(byte[] value)
+        {
+            if (value.Length != 4)
+            {
+                throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(value)} must contain exactly 4 bytes");
+            }
+
+            WriteHex(value[0], value[1], value[2], value[3]);
+        }
+
+        /// <summary>
+        /// Write 4 characters (from left to right)
+        /// </summary>
+        /// <param name="first">first character</param>
+        /// <param name="second">second character</param>
+        /// <param name="third">third character</param>
+        /// <param name="fourth">fourth character</param>
+        public void WriteChars(char first, char second, char third, char fourth)
+            => Write(Font.GetChar(first), Font.GetChar(second), Font.GetChar(third), Font.GetChar(fourth));
+
+        /// <summary>
+        /// Write integer value as decimal digits (from left to right)
+        /// </summary>
+        /// <param name="value">integer value</param>
+        /// <param name="alignment">alignment on display (left or right, right is default)</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="value"/> must be between -999..9999</exception>
+        public void WriteInt(int value, Alignment alignment = Alignment.Right)
+        {
+            if (value > 9999 || value < -999)
+            {
+                throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(value)} must be between -999..9999");
+            }
+
+            WriteString(value.ToString(), alignment);
+        }
+
+        /// <summary>
+        /// Write string value to display
+        /// </summary>
+        /// <param name="value">value to display, max 4 characters, or 5 characters if the 3rd character is ':' (this also turns on the center colon)</param>
+        /// <remarks>Unsupported characters will be replaced by whitespace</remarks>
+        /// <param name="alignment">alignment on display (left or right, right is default)</param>
+        /// <exception cref="ArgumentException"><paramref name="value"/>[2] must be a ':'</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="value"/> can contain maximum 5 characters</exception>
+        public void WriteString(string value, Alignment alignment = Alignment.Left)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                Clear();
+                return;
+            }
+
+            switch (value.Length)
+            {
+                case 4:
+                    WriteChars(value[0], value[1], value[2], value[3]);
+                    break;
+                case 5:
+                    if (value[2] != ':')
+                    {
+                        throw new ArgumentException(nameof(value), $"{nameof(value)}[2] must be a ':'");
+                    }
+                    Dots = Dot.CenterColon;
```

Comment:
when you write:
"12:34" and then on next call "1234" will center colon be erased? I currently cannot see any logic which would remove it

Reply:
It will not, it was a conscious decision, but it could go either way. The whole "12:34" branch is there because I mostly used this code to display time and this was a good shorthand.

Reply:
Although the first if that checks for empty input does clear the dots too. I'll make it right.